File: space-game/Core.Tests/Data/DistanceTestData.vb
Imports Core.Classes
Imports NUnit.Framework

Namespace Data
    Public Class DistanceTestData

        Public Shared Function Equals_WhenComparing_ReturnsExpected_Data() As List(Of TestCaseData)
            Dim nullDistance As Distance = Nothing
            Dim d = Distance.FromKilometers(12345)
            Dim copyOfD = d
            Dim notSame = Distance.FromKilometers(123456)

            Return New List(Of TestCaseData) From {
                New TestCaseData(d, nullDistance, False),
                New TestCaseData(d, copyOfD, True),
                New TestCaseData(d, notSame, False)
            }
        End Function

    End Class
End Namespace

File: space-game/Core.Tests/Data/DistanceTestData.cs
using Core.Classes;
using NUnit.Framework;
using System.Collections.Generic;

namespace Core.Tests.Data
{
    public class DistanceTestData
    {

        public static List<TestCaseData> Equals_WhenComparing_ReturnsExpected_Data()
        {
            Distance nullDistance = null;
            Distance d = Distance.FromKilometers(12345);
            Distance copyOfD = d;
            Distance notSame = Distance.FromKilometers(123456);

            return new List<TestCaseData> {
                new TestCaseData(d, nullDistance, false),
                new TestCaseData(d, copyOfD, true),
                new TestCaseData(d, notSame, false)
            };
        }

    }
}
